File: MedLabO.Models/MedicinskoOsoblje.cs

namespace MedLabO.Models
{
    public class MedicinskoOsoblje : ApplicationUser
    {
        public string? Ime { get; set; }
        public string? Prezime { get; set; }
        public bool IsActive { get; set; }
        public DateTime DTZaposlenja { get; set; }
        public DateTime? DTPrekidRadnogOdnosa { get; set; }
        public int? SpolID { get; set; }
        public virtual Spol? Spol { get; set; }
        public int? ZvanjeID { get; set; }
        public virtual Zvanje? Zvanje { get; set; }
    }
}

File: MedLabO UI/medlabo_desktop/lib/models/uposlenik/medicinsko_osoblje.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:medlabo_desktop/models/spol/spol.dart';
import 'package:medlabo_desktop/models/zvanje/zvanje.dart';

part 'medicinsko_osoblje.g.dart';

@JsonSerializable()
class MedicinskoOsoblje {
  String? ime;
  String? prezime;
  bool? isActive;
  String? dtZaposlenja;
  String? dtPrekidRadnogOdnosa;
  Spol? spol;
  Zvanje? zvanje;
  String? id;
  String? userName;
  String? email;
  String? phoneNumber;

  MedicinskoOsoblje(
      {this.ime,
      this.prezime,
      this.isActive,
      this.dtZaposlenja,
      this.dtPrekidRadnogOdnosa,
      this.spol,
      this.zvanje,
      this.id,
      this.userName,
      this.email,
      this.phoneNumber});

  factory MedicinskoOsoblje.fromJson(Map<String, dynamic> json) =>
      _$MedicinskoOsobljeFromJson(json);

  Map<String, dynamic> toJson() => _$MedicinskoOsobljeToJson(this);
}
